### ISSUE TITLE
how to use it with single article as input and pretrained model outputs summary?

### ISSUE THREAD

**Issue by bung87** [ChenRocks/fast_abs_rl#12]

**Comment by ChenRocks** [comment]
Thanks for you interest in this repository! However, could you elaborate more detail about the issue?

Note that is repository is more of a research code base than a general summarization API. It is most likely that we will not support your use case. However, feel free to fork and add your own enhancement (and cite our paper when appropriate :) ).

**Comment by bung87** [comment]
I read the `decode_full_model.py`,it seems it is used for validation or test purpose, and I read the readme it doesn't point out how to summarize a document.I want to know how to use it as other summarize project's general usage. string of single document -> string of summary.

**Comment by ChenRocks** [comment]
It is not currently supported (and probably won't ever be).

You can try to do it in 2 ways:
1. preprocess your data into the same dataset form
2. incorporate the necessary API (including PTBTokenizer) into decode_full_model.py

Feel free to submit a pull request if you add this functionality.

**Comment by bung87** [comment]
thanks for your response,I'll see what I can do on this.

**Comment by ChenRocks** [comment]
I'm closing the issue now since there's no bug in the code. Feel free to discuss the implementation in this thread.

**Comment by bung87** [comment]
I don't get the point,my question is single article as input and outputs summary, this function takes articles and abstracts outputs a Json file, what does it to with my question?

**Comment by bung87** [comment]
so you split out the preparation and save the outputs for decode usage, does it means it always need a set of data to preparation and decode, that's why ChenRocks says "It is not currently supported" the feature in my question?

**Comment by ChenRocks** [comment]
Thanks @Pked01 for providing the short script! Some minor issues worth notice here: the original preprocessing is based on CoreNLP PTBTokenizer so there could be quality issues on the summary due to train-test mismatch on preprocessing. Also the fact that when testing on a new dataset, there's no guarantee on accuracy (ROUGE).
@bung87 you can still use a single article to create a single json file with this approach to get what you want. However, due to above accuracy reason I currently have no plan to provide such an API in this repository.

**Comment by bung87** [comment]
Thanks guys! It helps me out ,it generates summary loss word's case.I think I will use non-deep-learning technic for now.

**Comment by Pked01** [comment]
@ChenRocks one doubt from my side...
I have seen very similar data preparation method in some other summarization method..pointer generator is one of them,,,
is it a standard approach...?
Or is there any particular advantages of using this method(using StanfordNLP, <s>sentence</s> , syntax etc.)?

**Comment by ChenRocks** [comment]
The pointer generator work you mentioned [(See et al, 2017)](https://arxiv.org/abs/1704.04368) is the first work standardizing the non-anonymized CNN/DM dataset. Since this is a research work, we followed their [preprocessing steps](https://github.com/abisee/cnn-dailymail) for a fair comparison and evaluation.

If you have your own training dataset, you can use whatever tokenization you like. Or you can re-tokenize the CNN/DM training data and re-train the model using a different tokenizer. The idea is to keep the training and testing preprocessing consistent.
